units/provisioning_user_data_done.go

```diff
@@ -111,13 +111,6 @@ func (j *userDataDoneJob) Run(ctx context.Context) {
 				"job":     j.ID(),
 			}))
 			j.AddError(err)
-			j.AddError(j.host.SetStatus(evergreen.HostProvisionFailed, evergreen.User,
```

Comment:
I think failures should get marked for termination at some point. otherwise we end up retrying forever...

Reply:
they eventually get cleaned up
https://github.com/evergreen-ci/evergreen/blob/46c6fcb94d0f0a001a029bff76cf2ada4e71f6f7/model/host/host.go#L1446

Reply:
it looks like these hosts never get cleaned up based on that as these are userdata hosts that are taking too long to provision but will never run tasks.

Reply:
I was understanding that if a userdata host isn't running a task and hasn't run a task before it will get cleaned up.

Reply:
that's not how I read that query.

Reply:
I get it now.
I put it back in. I thought it might be good to also add decommissioning for non-userdata hosts too.